### ISSUE TITLE
"Expected a single 'From' header" when trying to restore a Google Chat backup

### ISSUE THREAD

**Issue by jkeen** [jay0lee/got-your-back#73]
I'm getting this error when restoring, and in digging through the SQLite databases, I figured out the message it started happening on, and it's an old-style google chat. Must be right on the date I switched to Hangouts. 

Hangouts chats don't error (but don't restore in a useful way, either), but the old chats completely crash the restore process. The error makes sense, but I'm not sure what to do about it. 

This is what one of those files looks like. 

```json
sender {
  id {
    unique_id: "116702751922476051888"
    obfuscated_gaia_id: "116702751922476051888"
  }
  full_name: "Bob Sacamano"
}
conversation_id: "UgzQVz1lat0ak63hkk54AaABAQ"
timestamp: 1372524339230852
event_id: "7-H0Z7-2xai70pr8U8voM0"
chat_message {
  segment {
    text: "yup"
  }
}
is_off_the_record: false```

**Comment by samsar777** [comment]
I have seen few emial causing same errors at my mailbox. There were two reasons:
- two From headers
- no From header

Once I have identified them (I have modified the code, so it prints to the console which message is being restored for both batch restore and single big message), I have changed message name, so it was skipped during import. For few that I really need, I have forwarded them from old to new email account. No other idea how to solve this.

Hope this helps.

**Comment by ratio91** [comment]
I experienced the same error after having sucessfully restored more than 25,000 mails (and hangouts chat messages).

`Expected a single 'From header'`

@samsar777 could you please tell me what parts of the code you exactly modified in order to get it working? Thanks!

**Comment by jay0lee** [comment]
This should be fixed in 0.43.
